src/clj_gatling/simulation.clj

```diff
@@ -123,62 +137,90 @@
                 (nil? step')
                 (and skip-next-after-failure?
                      (request-failed? result)))
-          (do
-            (when post-hook
-              (post-hook context))
-            (>! result-channel (conj results (clean-result result))))
+          (>! result-channel [(conj results (clean-result result)) context])
```

Comment:
The `post-hook` has been moved out of the run-once so that longer post-hooks (e.g. cleaning up test data) don't show as an active concurrent scenario against the system under test.

There might still be slowness within the step functions themselves, generating the context for the next step, but there's nothing we can do about that. It's also less likely than slowness on setup/teardown.